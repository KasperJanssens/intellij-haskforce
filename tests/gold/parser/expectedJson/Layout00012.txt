[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":9,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":20}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":8},"srcInfoPoints":[]},"Layout00012"]},null,null]},[],[],[{"TypeDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":67,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":5,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":19}]},{"DHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":6},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":6},"srcInfoPoints":[]},"BoundedInt"]},[{"UnkindedVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},"srcInfoPoints":[]},"a"]}]}]]},{"TyTuple":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":67,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":21},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":21},{"srcSpanFilename":"A.hs","srcSpanEndColumn":37,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":36},{"srcSpanFilename":"A.hs","srcSpanEndColumn":67,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":66}]},"Boxed",[{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":36,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22},"srcInfoPoints":[]},"BoundedSuper"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":36,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":35},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":36,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":35},"srcInfoPoints":[]},"a"]}]}]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":66,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":38},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":50,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":38},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":50,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":38},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":50,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":38},"srcInfoPoints":[]},"BoundedSuper"]}]}]},{"TyParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":66,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":51},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":51},{"srcSpanFilename":"A.hs","srcSpanEndColumn":66,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":65}]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":65,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":52},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":63,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":52},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":63,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":52},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":63,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":52},"srcInfoPoints":[]},"UnsignedRep"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":65,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":64},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":65,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":64},"srcInfoPoints":[]},"a"]}]}]}]}]}]]}]},{"ClassDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":79,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":6,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1}]},{"CxTuple":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":64,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":10},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":10},{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":16},{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":23},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":34},{"srcSpanFilename":"A.hs","srcSpanEndColumn":47,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":46},{"srcSpanFilename":"A.hs","srcSpanEndColumn":61,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":60},{"srcSpanFilename":"A.hs","srcSpanEndColumn":64,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":62}]},[{"ClassA":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":11},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":11},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":11},"srcInfoPoints":[]},"Ord"]}]},[{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":15},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":15},"srcInfoPoints":[]},"a"]}]}]]},{"ClassA":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},"srcInfoPoints":[]},"Num"]}]},[{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":22},"srcInfoPoints":[]},"a"]}]}]]},{"ClassA":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":25},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":32,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":25},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":32,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":25},"srcInfoPoints":[]},"Bounded"]}]},[{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":33},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":33},"srcInfoPoints":[]},"a"]}]}]]},{"ClassA":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":46,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":36},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":44,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":36},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":44,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":36},"srcInfoPoints":[]},"Integral"]}]},[{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":46,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":45},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":46,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":45},"srcInfoPoints":[]},"a"]}]}]]},{"ClassA":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":60,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":48},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":58,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":48},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":58,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":48},"srcInfoPoints":[]},"FiniteBits"]}]},[{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":60,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":59},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":60,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":59},"srcInfoPoints":[]},"a"]}]}]]}]]},{"DHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":77,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":65},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":77,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":65},"srcInfoPoints":[]},"BoundedSuper"]},[{"UnkindedVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":79,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":78},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":79,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":78},"srcInfoPoints":[]},"a"]}]}]]},[],null]}]]},[{"Comment":[false,{"srcSpanFilename":"A.hs","srcSpanEndColumn":33,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":1}," | Super class to 'BoundedInt'"]}]]