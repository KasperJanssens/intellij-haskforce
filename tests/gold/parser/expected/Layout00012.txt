Haskell File
  HaskellModuledeclImpl(MODULEDECL)
    PsiElement(HaskellTokenType.module)('module')
    HaskellQconidImpl(QCONID)
      HaskellConidImpl(CONID)
        PsiElement(HaskellTokenType.conidRegexp)('Layout00012')
    PsiElement(HaskellTokenType.where)('where')
  HaskellBodyImpl(BODY)
    HaskellTypedeclImpl(TYPEDECL)
      PsiElement(HaskellTokenType.type)('type')
      HaskellTypeeImpl(TYPEE)
        HaskellAtypeImpl(ATYPE)
          HaskellOqtyconImpl(OQTYCON)
            HaskellQtyconImpl(QTYCON)
              HaskellTyconImpl(TYCON)
                HaskellConidImpl(CONID)
                  PsiElement(HaskellTokenType.conidRegexp)('BoundedInt')
        HaskellAtypeImpl(ATYPE)
          HaskellTyvarImpl(TYVAR)
            HaskellVaridImpl(VARID)
              PsiElement(HaskellTokenType.varidRegexp)('a')
      PsiElement(HaskellTokenType.=)('=')
      HaskellTypeeImpl(TYPEE)
        HaskellAtypeImpl(ATYPE)
          PsiElement(HaskellTokenType.()('(')
          HaskellCtypeImpl(CTYPE)
            HaskellTypeeImpl(TYPEE)
              HaskellAtypeImpl(ATYPE)
                HaskellOqtyconImpl(OQTYCON)
                  HaskellQtyconImpl(QTYCON)
                    HaskellTyconImpl(TYCON)
                      HaskellConidImpl(CONID)
                        PsiElement(HaskellTokenType.conidRegexp)('BoundedSuper')
              HaskellAtypeImpl(ATYPE)
                HaskellTyvarImpl(TYVAR)
                  HaskellVaridImpl(VARID)
                    PsiElement(HaskellTokenType.varidRegexp)('a')
          PsiElement(HaskellTokenType.,)(',')
          HaskellCtypeImpl(CTYPE)
            HaskellTypeeImpl(TYPEE)
              HaskellAtypeImpl(ATYPE)
                HaskellOqtyconImpl(OQTYCON)
                  HaskellQtyconImpl(QTYCON)
                    HaskellTyconImpl(TYCON)
                      HaskellConidImpl(CONID)
                        PsiElement(HaskellTokenType.conidRegexp)('BoundedSuper')
              HaskellAtypeImpl(ATYPE)
                PsiElement(HaskellTokenType.()('(')
                HaskellCtypeImpl(CTYPE)
                  HaskellTypeeImpl(TYPEE)
                    HaskellAtypeImpl(ATYPE)
                      HaskellOqtyconImpl(OQTYCON)
                        HaskellQtyconImpl(QTYCON)
                          HaskellTyconImpl(TYCON)
                            HaskellConidImpl(CONID)
                              PsiElement(HaskellTokenType.conidRegexp)('UnsignedRep')
                    HaskellAtypeImpl(ATYPE)
                      HaskellTyvarImpl(TYVAR)
                        HaskellVaridImpl(VARID)
                          PsiElement(HaskellTokenType.varidRegexp)('a')
                PsiElement(HaskellTokenType.))(')')
          PsiElement(HaskellTokenType.))(')')
    PsiComment(HaskellTokenType.comment)('-- | Super class to 'BoundedInt'\n')
    HaskellCppImpl(CPP)
      PsiElement(HaskellTokenType.CPPIF)('#if defined(__GLASGOW_HASKELL__) && __GLASGOW_HASKELL__ >= 708')
    HaskellClassdeclImpl(CLASSDECL)
      PsiElement(HaskellTokenType.class)('class')
      HaskellCtypeImpl(CTYPE)
        HaskellContextImpl(CONTEXT)
          HaskellAtypeImpl(ATYPE)
            PsiElement(HaskellTokenType.()('(')
            HaskellCtypeImpl(CTYPE)
              HaskellTypeeImpl(TYPEE)
                HaskellAtypeImpl(ATYPE)
                  HaskellOqtyconImpl(OQTYCON)
                    HaskellQtyconImpl(QTYCON)
                      HaskellTyconImpl(TYCON)
                        HaskellConidImpl(CONID)
                          PsiElement(HaskellTokenType.conidRegexp)('Ord')
                HaskellAtypeImpl(ATYPE)
                  HaskellTyvarImpl(TYVAR)
                    HaskellVaridImpl(VARID)
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.,)(',')
            HaskellCtypeImpl(CTYPE)
              HaskellTypeeImpl(TYPEE)
                HaskellAtypeImpl(ATYPE)
                  HaskellOqtyconImpl(OQTYCON)
                    HaskellQtyconImpl(QTYCON)
                      HaskellTyconImpl(TYCON)
                        HaskellConidImpl(CONID)
                          PsiElement(HaskellTokenType.conidRegexp)('Num')
                HaskellAtypeImpl(ATYPE)
                  HaskellTyvarImpl(TYVAR)
                    HaskellVaridImpl(VARID)
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.,)(',')
            HaskellCtypeImpl(CTYPE)
              HaskellTypeeImpl(TYPEE)
                HaskellAtypeImpl(ATYPE)
                  HaskellOqtyconImpl(OQTYCON)
                    HaskellQtyconImpl(QTYCON)
                      HaskellTyconImpl(TYCON)
                        HaskellConidImpl(CONID)
                          PsiElement(HaskellTokenType.conidRegexp)('Bounded')
                HaskellAtypeImpl(ATYPE)
                  HaskellTyvarImpl(TYVAR)
                    HaskellVaridImpl(VARID)
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.,)(',')
            HaskellCtypeImpl(CTYPE)
              HaskellTypeeImpl(TYPEE)
                HaskellAtypeImpl(ATYPE)
                  HaskellOqtyconImpl(OQTYCON)
                    HaskellQtyconImpl(QTYCON)
                      HaskellTyconImpl(TYCON)
                        HaskellConidImpl(CONID)
                          PsiElement(HaskellTokenType.conidRegexp)('Integral')
                HaskellAtypeImpl(ATYPE)
                  HaskellTyvarImpl(TYVAR)
                    HaskellVaridImpl(VARID)
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.,)(',')
            HaskellCtypeImpl(CTYPE)
              HaskellTypeeImpl(TYPEE)
                HaskellAtypeImpl(ATYPE)
                  HaskellOqtyconImpl(OQTYCON)
                    HaskellQtyconImpl(QTYCON)
                      HaskellTyconImpl(TYCON)
                        HaskellConidImpl(CONID)
                          PsiElement(HaskellTokenType.conidRegexp)('FiniteBits')
                HaskellAtypeImpl(ATYPE)
                  HaskellTyvarImpl(TYVAR)
                    HaskellVaridImpl(VARID)
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.))(')')
        PsiElement(HaskellTokenType.=>)('=>')
        HaskellCtypeImpl(CTYPE)
          HaskellTypeeImpl(TYPEE)
            HaskellAtypeImpl(ATYPE)
              HaskellOqtyconImpl(OQTYCON)
                HaskellQtyconImpl(QTYCON)
                  HaskellTyconImpl(TYCON)
                    HaskellConidImpl(CONID)
                      PsiElement(HaskellTokenType.conidRegexp)('BoundedSuper')
            HaskellAtypeImpl(ATYPE)
              HaskellTyvarImpl(TYVAR)
                HaskellVaridImpl(VARID)
                  PsiElement(HaskellTokenType.varidRegexp)('a')
    HaskellCppImpl(CPP)
      PsiElement(HaskellTokenType.#endif)('#endif')